### PR TITLE
ENH: Use override for destructors and = default where appropriate

### PR DESCRIPTION
**Pull request by dzenanz**
This only cleans up files which don't get removed in #161.

### CODE REVIEW

**Comment by dzenanz** [comment]
Ping @tbirdso.

**Comment by dzenanz** [comment]
@pranjalSahu could also review.

File: include/itkBlockMatchingMultiResolutionIterationObserver.hxx [hunk 1]
```diff
@@ -48,7 +48,7 @@ MultiResolutionIterationObserver<TMultiResolutionMethod>::MultiResolutionIterati
 
 
 template <typename TMultiResolutionMethod>
-MultiResolutionIterationObserver<TMultiResolutionMethod>::~MultiResolutionIterationObserver()
+MultiResolutionIterationObserver<TMultiResolutionMethod>::~MultiResolutionIterationObserver() override
```
Comment (tbirdso) [inline include/itkBlockMatchingMultiResolutionIterationObserver.hxx:51]: From [CDash](https://open.cdash.org/viewBuildError.php?buildid=7665861):

```
include/itkBlockMatchingMultiResolutionIterationObserver.hxx:51:95: error: virt-specifiers in 'MultiResolutionIterationObserver' not allowed outside a class definition
```

I don't think the `override` is need in the implementation here if it is included in the declaration in `itkBlockMatchingMultiResolutionIterationObserver.h `

**Review by tbirdso — CHANGES_REQUESTED** [review]
One fix for CDash failures, otherwise LGTM

File: include/itkFFTWCommonExtended.h [hunk 2]
```diff
@@ -45,8 +45,8 @@ class ComplexToComplexProxy
 {
   // empty -- only double and float specializations work
 protected:
-  ComplexToComplexProxy(){};
-  ~ComplexToComplexProxy(){};
+  ComplexToComplexProxy() = default;
+  ~ComplexToComplexProxy() override = default;
```
Comment (tbirdso) [inline include/itkFFTWCommonExtended.h:49]: [CDash result](include/itkFFTWCommonExtended.h:49:28: error: only virtual member functions can be marked 'override')

Looks like `ComplexToComplexProxy` does not inherit from a superclass and should not use `override`

**Review by tbirdso — COMMENTED** [review]
LGTM once CI issues are resolved

**Comment by PranjalSahu** [comment]
LGTM

**Pull request merged**